test/client/Mutation.test.tsx

```diff
@@ -1476,107 +1476,210 @@ it('errors when changing from mutation to a subscription', done => {
   console.log = errorLogger; // tslint:disable-line
 });
 
-it('does not update state after receiving data after it has been unmounted', done => {
-  let success = false;
-  let original = console.error;
-  console.error = jest.fn();
-  const checker = () => {
-    setTimeout(() => {
-      expect(console.error).not.toHaveBeenCalled();
-      console.error = original;
-      success = true;
-      done();
-    }, 100);
-  };
-
-  class Component extends React.Component {
-    state = {
-      called: false,
+describe('after it has been unmounted', () => {
+  it('calls the onCompleted prop after the mutation is complete', done => {
```

Comment:
I added a new describe block for when Mutation is unmounted, and two new tests to make sure Mutation is calling `onCompleted` and `onError` after it has been unmounted. This should cover all changes introduced in this PR.

```
  after it has been unmounted
    ✓ calls the onCompleted prop after the mutation is complete (109ms)
    ✓ calls the onError prop if the mutation encounters an error (109ms)
    ✓ does not update state after receiving data (111ms)
    ✓ does not update state after receiving error (123ms)
```